src/libstd/sys/unix/thread.rs

```diff
@@ -199,6 +201,22 @@ pub mod guard {
         current().map(|s| s as *mut libc::c_void)
     }
 
+    #[cfg(target_os = "freebsd")]
+    unsafe fn get_stack_start() -> Option<*mut libc::c_void> {
+        let mut ret = None;
+        let mut attr: libc::pthread_attr_t = ::mem::zeroed();
+        assert_eq!(libc::pthread_attr_init(&mut attr), 0);
+        if libc::pthread_attr_get_np(libc::pthread_self(), &mut attr) == 0 {
+            let mut stackaddr = ::ptr::null_mut();
+            let mut stacksize = 0;
+            assert_eq!(libc::pthread_attr_getstack(&attr, &mut stackaddr,
+                                                   &mut stacksize), 0);
+            ret = Some(stackaddr);
+        }
+        assert_eq!(libc::pthread_attr_destroy(&mut attr), 0);
+        ret
+    }
```

Comment:
This looks pretty similar to `get_stack_start` with Linux/Android/NetBSD below, perhaps the two could share code? If the only difference is the name of the function `pthread_getattr_np`, then some renaming-on import should suffice I think.